### DISCUSSION TITLE
Details about rounding

### DISCUSSION THREAD

**Discussion by jkanner** [gwosc-tutorial/gwosc-catalog#77] [General]
The `best` value of a parameter is displayed in the Event Portal using this template tag:

```
{{ param.best|format_value:param.sigfigs }}
```
where param.sigfigs corresponds to the `decimal_places` parameter.

**Comment by jkanner** [comment]
The `format_value` method is python code that looks like this:

```
def format_value(value, sigfigs):
    if value is None:
        result = ""
    elif (abs(value) >= 10 ** (-sigfigs)) or (value == 0.0):
        result = "{:.{prec}f}".format(value, prec=sigfigs)
    else:
        result = f"{value:.1e}"
    return result
```

**Comment by jkanner** [comment]
I will try to put words to this:

* In most cases, values are displayed as floating point numbers with the `decimal_places` corresponding to how many digits are displayed after the decimal place.  Setting `decimal_places` to 0 results in printing a whole number

* The special case is if the number of decimals to display is not enough for the value of the number.  For example, if we have a number 0.001, but ask to display to only 1 decimal place.  This would round the number to dispaly `0.0`, which might not be appropriate.  In this case, the code instead displays the value in scientific notation with 1 decimal place.  In this example, that is `1.0e-3`

**Comment by martinberoiz** [comment]
I found this related discussion (it's rather long), about python formatting floats into strings. As a disclaimer, I don't know all the details about our rounding or chopping figures in the table. 

https://discuss.python.org/t/formatting-numbers-to-string-with-r-formatting-for-rounded-precision/68224
